app/src/main/res/layout/fragment_plant_detail.xml

```diff
@@ -119,7 +119,8 @@
             android:layout_gravity="center_vertical|start"
             android:layout_margin="@dimen/fab_margin"
             android:tint="@android:color/white"
-            app:minusIf="@{viewModel.isPlanted}"
+            android:onClick="@{(v) -> viewModel.toggleAddOrRemove(v)}"
```

Comment:
1. *Try* to decouple the `View` here. Actually, the `ViewModel` should not know the `View`.
` android:onClick="@{(_) -> viewModel.toggleAddOrRemove()}"`
2. Show of `Snackbar` should be still kept in `Fragment` which is just `UI` biz.  
     Tip in `PlantDetailViewModel` :
```
             class PlantDetailViewModel {
                         val  notifyToggleAddOrRemove: LiveData
                         fun toggleAddOrRemove() {
                             if() ...notifyToggleAddOrRemove.post
                             else() ... notifyToggleAddOrRemove.post
                         }
             }
          
```

Reply:
@XinyueZ Cool, I'm agree with you. I have tried to decouple view from ViewModel before but not successful :sweat:. Thanks for your Tip!